src/Rules/Properties/PropertyInClassRule.php

```diff
@@ -81,6 +93,54 @@ public function processNode(Node $node, Scope $scope): array
 			];
 		}
 
+		if ($node->isPrivate()) {
+			if ($node->isFinal()) {
+				return [
+					RuleErrorBuilder::message('Property cannot be both final and private.')
+						->nonIgnorable()
+						->identifier('property.finalPrivate')
+						->build(),
+				];
+			}
+
+			foreach ($node->getHooks() as $hook) {
+				if (!$hook->isFinal()) {
+					continue;
+				}
+
+				return [
+					RuleErrorBuilder::message('Property cannot be both final and private.')
```

Comment:
This message should say it's about the hook. "Private property cannot have a final hook" or something like that.

Reply:
thanks - I am still learning this new property hooks thing :-)